### PR TITLE
Add Patch Request and refactored GetPayment to be internal function

### PR DESCRIPTION
**Pull request by jphillips2121**
- Added a new Patch Request which can update only the payment_method and status fields. 
- Refactored the GetPayment and GetCosts functions so they can be called from internal function ready for later POST request functionality for external pay session.

Resolves: CPS-208

### Type of change

* [ ] Bug fix
* [X] New feature
* [ ] Breaking change

### Pull request checklist

* [X] I have added unit tests for new code that I have added
* [X] I have added/updated functional tests where appropriate
* [X] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/go.md)

### CODE REVIEW

File: dao/mongo.go [hunk 1]
```diff
@@ -64,3 +66,28 @@ func (m *Mongo) GetPaymentResource(id string) (*models.PaymentResource, error) {
 
 	return &resource, err
 }
+
+func (m *Mongo) PatchPaymentResource(id string, paymentUpdate *models.PaymentResourceData) error {
```
Comment (emead) [inline dao/mongo.go]: Exported method should have comment

File: dao/mongo.go [hunk 2]
```diff
@@ -64,3 +66,28 @@ func (m *Mongo) GetPaymentResource(id string) (*models.PaymentResource, error) {
...
+	// Patch only these fields
+	if paymentUpdate.PaymentMethod != "" {
+		patchUpdate["data.payment_method"] = paymentUpdate.PaymentMethod
+	}
+	if paymentUpdate.Status != "" {
+		patchUpdate["data.status"] = paymentUpdate.Status
+	}
+
+	updateCall := bson.M{"$set": patchUpdate}
+	c.UpdateId(id, updateCall)
```
Comment (emead) [inline dao/mongo.go]: error return value not checked

File: models/payment.go [hunk 3]
```diff
@@ -25,7 +25,7 @@ type PaymentResourceData struct {
 	CreatedBy               CreatedBy      `json:"created_by"                          bson:"created_by"`
 	Description             string         `json:"description"                         bson:"description"`
 	Links                   Links          `json:"links"                               bson:"links"`
-	PaymentMethod           string         `json:"payment_method,omitempty"            bson:"payment_method,omitempty"`
+	PaymentMethod           string         `json:"payment_method"                      bson:"payment_method"`
```
Comment (emead) [inline models/payment.go]: According to the spec, `payment_method` is an optional field, so why has `omitempty` been removed?

File: dao/mongo.go [hunk 4]
```diff
@@ -3,6 +3,8 @@ package dao
 import (
 	"fmt"
 
+	"github.com/globalsign/mgo/bson"
```
Comment (emead) [inline dao/mongo.go]: Please group with the existing globalsign import

File: models/payment.go (hunk 3 above)
Comment (jphillips2121) [reply models/payment.go]: The patch request won't work updating the field unless there is already a value present

Comment (jphillips2121) [reply models/payment.go]: Following discussion the json can be left as `omitempty` whilst it's removed for the bson and this should still work whilst adhering to the spec

**Pull request merged**